src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h

```diff
@@ -644,22 +644,22 @@ typedef VOID (*RtcOnPictureLoss)(UINT64);
  *
  * Reference: https://www.w3.org/TR/webrtc/#dom-rtcdatachannel-onmessage
  */
-typedef VOID (*RtcOnMessage)(UINT64, BOOL, PBYTE, UINT32);
+struct __RtcDataChannel;
+typedef VOID (*RtcOnMessage)(UINT64, struct __RtcDataChannel*, BOOL, PBYTE, UINT32);
 
 /**
  * RtcOnOpen is fired when the DataChannel has opened
  *
  * Reference: https://www.w3.org/TR/webrtc/#dom-rtcdatachannel-onopen
  */
-typedef VOID (*RtcOnOpen)(UINT64);
+typedef VOID (*RtcOnOpen)(UINT64, struct __RtcDataChannel*);
```

Comment:
My main concern here is the backward compatibility. If we implement this change then we need to rev the major version. We could have a separate callback with a different name that would ensure the compile-time compatibility but it would really be ugly but I just don't see any other solution. 

Another proposal is to redefine the minor version update to indicate that in case of the minor version update there might be some compile time compatibility issues that can be trivial to solve in the application code. I would vote for the second. The reason is that this is still a very active development codebase on a dynamic technology standard that morphs continuously and vamping a major version with small incremental implementation/changes like these makes no sense.

Please discuss/debage.

Reply:
Yeah, I hear you. It's a breaking change. As I could not come up with another workaround, I will stick with my fork with the hope that you would role out major version update.

If I were you, I wouldn't be so concerned about advancing the major version rapidly. I'd rather communicate what you are releasing to users by strictly following semantic versioning. (2c)

Reply:
I agree, revving the major version is not needed in this case. Let us make  few minor changes per the comments and we will push another pr as co-authors

Reply:
That's not what I meant tho. I was saying, "don't be afraid to advance the major version number, and you should increase the version number when there's a breaking change so that users will know what he/she can take appropriate action".  But, It's your call.

Reply:
The issue is that this codebase is still undergoing active development. For example, we will be adding many new structures to the code as we progress and perhaps changing some minor callbacks. Revving up a major version is just going to be more confusing. I think for these types of changes my vote would be to rev the minor version (not the revision though). At any rate, let's get these changes in first. When cutting the release we can properly word the release notes with this.